packages/terra-popup/src/Popup.jsx

```diff
@@ -0,0 +1,370 @@
+    this.contentNode.classList.remove(PopupContent.oppositePositionClasses[position]);
+
+    this.arrowNode.classList.add(PopupArrow.positionClasses[position]);
+    this.contentNode.classList.add(PopupContent.positionClasses[position]);
+
+    if (isVerticalAttachment) {
+      this.arrowNode.style.left = Popup.leftOffset(targetBounds, contentBounds, PopupArrow.arrowSize, this.offset, this.attachment);
+    } else {
+      this.arrowNode.style.top = Popup.topOffset(targetBounds, contentBounds, PopupArrow.arrowSize);
+    }
+  }
+
+  setArrowNode(node) {
+    this.arrowNode = node;
+  }
+
+  setContentNode(node) {
+    this.contentNode = node;
+  }
+
+  handleTetherRepositioned(event, targetBounds, contentBounds) {
+    if (this.arrowNode && this.contentNode) {
+      this.setArrowPosition(targetBounds, contentBounds);
+    }
+  }
+
+  createPopupContent(boundingFrame) {
+    const parsedDimenions = Popup.parseStringPair(this.props.contentDimensions);
+    const boundsProps = {
+      contentWidth: BASE_WIDTH * DIMENSIONS_MAP[parsedDimenions.horizontal],
+      contentHeight: BASE_HEIGHT * DIMENSIONS_MAP[parsedDimenions.vertical],
+    };
+
+    if (boundingFrame) {
+      boundsProps.contentHeightMax = boundingFrame.clientHeight;
+      boundsProps.contentWidthMax = boundingFrame.clientWidth;
+    } else {
+      boundsProps.contentHeightMax = window.innerHeight;
+      boundsProps.contentWidthMax = window.innerWidth;
+    }
+
+    let arrow;
+    let arrowPosition;
+    if (this.props.isArrowDisplayed && this.props.contentAttachment !== 'middle center') {
+      this.offset = Popup.getContentOffset(this.attachment, this.props.targetRef(), PopupArrow.arrowSize);
+      arrow = <PopupArrow className={this.props.classNameArrow} refCallback={this.setArrowNode} />;
+      arrowPosition = Popup.primaryArrowPosition(this.attachment);
+    }
+
+    return (
+      <PopupContent
+        {...boundsProps}
+        arrow={arrow}
+        arrowPosition={arrowPosition}
+        classNameInner={this.props.classNameContent}
+        closeOnEsc
+        closeOnOutsideClick
+        closeOnResize
+        isHeaderDisabled={this.props.isHeaderDisabled}
+        onRequestClose={this.props.onRequestClose}
+        refCallback={this.setContentNode}
+      >
+        {this.props.children}
+      </PopupContent>
+    );
+  }
+
+  render() {
+    const {
+      boundingRef,
+      children,
+      classNameArrow,
+      classNameContent,
+      classNameOverlay,
+      contentAttachment,
+      contentDimensions,
+      isArrowDisplayed,
+      isHeaderDisabled,
+      isOpen,
+      onRequestClose,
+      targetRef,
+      ...customProps
+    } = this.props; // eslint-disable-line no-unused-vars
+
+    let portalContent = children;
+    if (isOpen) {
+      this.attachment = Popup.parseStringPair(contentAttachment);
+
+      const boundingFrame = boundingRef ? boundingRef() : undefined;
```

Comment:
I am curious. In you're example, the `boundingRef` gets the parentNode. Are there instances where that the boundingRef will not be the parentNode?